pkg/render/manager_route_config.go

```diff
@@ -0,0 +1,396 @@
+	var tunnelTLSTerminatedRoutes []*tlsTerminatedRoute
+	var tunnelTLSPassThroughRoutes []*tlsPassThroughRoute
+
+	for _, route := range builder.tlsTerminatedRoutes {
+		if route.Spec.CABundle == nil && !route.Spec.AllowInsecureTLS {
+			return nil, fmt.Errorf("CABundle must be set if AllowInsecureTLS is false")
+		}
+
+		r := &tlsTerminatedRoute{
+			Destination:      route.Spec.Destination,
+			Path:             route.Spec.PathMatch.Path,
+			PathRegexp:       route.Spec.PathMatch.PathRegexp,
+			PathReplace:      route.Spec.PathMatch.PathReplace,
+			AllowInsecureTLS: route.Spec.AllowInsecureTLS,
+		}
+
+		if route.Spec.Target == operatorv1.TargetTypeUI {
+			uiTLSTerminatedRoutes = append(uiTLSTerminatedRoutes, r)
+			if route.Spec.Unauthenticated != nil {
+				r.Unauthenticated = *route.Spec.Unauthenticated
+			}
+		} else if route.Spec.Target == operatorv1.TargetTypeUpstreamTunnel {
+			tunnelTLSTerminatedRoutes = append(tunnelTLSTerminatedRoutes, r)
+		} else {
+			return nil, fmt.Errorf("unknown Target value %s", route.Spec.Target)
+		}
+
+		if route.Spec.CABundle != nil {
+			path, err := builder.mountConfigMapReference(route.Spec.CABundle.Name, route.Spec.CABundle.Key)
+			if err != nil {
+				return nil, err
+			}
+
+			r.CABundlePath = path
+		}
+
+		// Require that either both MTLSCert and MTLSKey are set or neither are.
+		if (route.Spec.MTLSCert != nil && route.Spec.MTLSKey == nil) || (route.Spec.MTLSKey != nil && route.Spec.MTLSCert == nil) {
+			return nil, fmt.Errorf("must set both MTLSCert and MTLSKey, or neither for TLS terminated route %s", route.Name)
+		}
+
+		if route.Spec.MTLSCert != nil {
+			path, err := builder.mountSecretReference(route.Spec.MTLSCert.Name, route.Spec.MTLSCert.Key)
+			if err != nil {
+				return nil, err
+			}
+			r.ClientCertPath = path
+
+			// At this point, if MTLSCert is set then MTLSKey will be set otherwise this if statement wouldn't be executed
+			// or an error would have already been returned.
+			path, err = builder.mountSecretReference(route.Spec.MTLSKey.Name, route.Spec.MTLSKey.Key)
+			if err != nil {
+				return nil, err
+			}
+			r.ClientKeyPath = path
+		}
+	}
+
+	for _, route := range builder.tlsPassThroughRoutes {
+		r := &tlsPassThroughRoute{
+			Destination: route.Spec.Destination,
+			ServerName:  route.Spec.SNIMatch.ServerName,
+		}
+		tunnelTLSPassThroughRoutes = append(tunnelTLSPassThroughRoutes, r)
+	}
+
+	routesData := map[string]string{}
+
+	if len(uiTLSTerminatedRoutes) > 0 {
+		jsonBytes, err := json.Marshal(uiTLSTerminatedRoutes)
+		if err != nil {
+			return nil, err
+		}
+
+		routesData[uiTLSTerminatedRoutesKey] = string(jsonBytes)
+	}
+
+	if len(tunnelTLSTerminatedRoutes) > 0 {
+		jsonBytes, err := json.Marshal(tunnelTLSTerminatedRoutes)
+		if err != nil {
+			return nil, err
+		}
+
+		routesData[upstreamTunnelTLSTerminatedRoutesKey] = string(jsonBytes)
+	}
+
+	if len(tunnelTLSPassThroughRoutes) > 0 {
+		jsonBytes, err := json.Marshal(tunnelTLSPassThroughRoutes)
```

Comment:
Given this is all going into a configmap / config file.. might be worth using `MarshalIndent` to make it human readable? Easier to debug with `kubectl get configmap -o yaml`

Reply:
Done.